bookkeeper-server/src/main/java/org/apache/bookkeeper/client/RegionAwareEnsemblePlacementPolicy.java

```diff
@@ -360,8 +361,12 @@ public PlacementResult<List<BookieId>> newEnsemble(int ensembleSize, int writeQu
                         effectiveMinRegionsForDurability, minNumRacksPerWriteQuorum);
                 TopologyAwareEnsemblePlacementPolicy nextPolicy = perRegionPlacement.get(
                         availableRegions.iterator().next());
-                return nextPolicy.newEnsemble(ensembleSize, writeQuorumSize, writeQuorumSize,
+
+                PlacementResult<List<BookieId>> placementResult = nextPolicy.newEnsemble(ensembleSize, writeQuorumSize, writeQuorumSize,
                         comprehensiveExclusionBookiesSet, ensemble, ensemble);
+                return PlacementResult.of(placementResult.getResult(),
+                        isEnsembleAdheringToPlacementPolicy(
```

Comment:
Here, we can simplify the logic. We already know there is only one region. 
Just need check the  result of `nextPolicy.newEnsemble`.
1. If the  result is PlacementPolicyAdherence.FAIL, return directly. 
2. If the result is  PlacementPolicyAdherence.MEETS_STRICT, then check whether minNumRacksPerWriteQuorum is more than 1. If it is, return PlacementPolicyAdherence.SOFT, otherwise return PlacementPolicyAdherence.MEETS_STRICT.